StRoot/StGmtClusterMaker/StGmtClusterMaker.cxx

```diff
@@ -0,0 +1,307 @@
+    else               { profPointer=profY[module]; histPointer=histY; }
+    for(UInt_t iTimeBin=0;iTimeBin<kGmtNumTimeBins;iTimeBin++) {
+      adc_buf=pStrip->getAdc(iTimeBin);
+      if(adc_buf>-999) adc+=adc_buf;
+    }
+    position=pStrip->getPosition();
+    int bin=profPointer->Fill(position,adc);
+    Double_t error=TMath::Sqrt(adc);
+    histPointer->Fill(position,adc);
+    histPointer->SetBinError(bin,error);
+  }
+  if (events < 5) return;
+  histX->Add(&profXold,-1.0); histY->Add(&profYold,-1.0);
+  
+  TF1 *fitX=0, *fitY=0;
+  fitX = FindPeaks(histX); fitY = FindPeaks(histY);
+  UInt_t idx[MAX_PEAKS], idy[MAX_PEAKS];
+  UInt_t nClusX=0, nClusY=0;
+  if(fitX) {
+    for(UInt_t i=0; i<fitX->GetParameter(0); i++) {
+      if (fitX->GetParameter(3*i+3) > 0.4) continue;
+      if (fitX->GetParameter(3*i+1) < 5.0) continue;
+      idx[nClusX] = i;
+      nClusX++;
+    }
+    if (nClusX) *profX[module]=profXold;      
+    if (Debug()) {LOG_INFO << "######XPEAKS found =" << fitX->GetParameter(0) << ", Clusters fitted =" << nClusX << endm;}
+  } else if (Debug()) {LOG_INFO << "######XNULL" <<  endm; }
+  if(fitY) {
+    for(UInt_t i=0; i<fitY->GetParameter(3*i+3); i++) {
+      if (fitY->GetParameter(3*i+3) > 0.4) continue;
+      if (fitY->GetParameter(3*i+1) < 5.0) continue;
+      idy[nClusY] = i;
+      nClusY++;
+    }
+    if (nClusY) *profY[module]=profYold;
+    if (Debug()) {LOG_INFO << "######YPEAKS found =" << fitY->GetParameter(0) << ", Clusters fitted =" << nClusY << endm;}
+  } else if (Debug()) {LOG_INFO << "######YNULL" << endm; }
+  for(UInt_t i=0; i < nClusX; i++) {
+    UInt_t nx = idx[i];
+    for(UInt_t j = 0; j < nClusY; j++) {
+      UInt_t ny = idy[j];
+      StGmtHit* newCluster = new StGmtHit(
+					  hits.getHitVec().size(),
+					  module, 
+					  TMath::Exp(fitX->GetParameter(3*nx+1)), // adcX
+					  TMath::Exp(fitY->GetParameter(3*ny+1)), // adcY
+					  fitX->GetParError(3*nx+1),              // error(adcX)
+					  fitY->GetParError(3*ny+1),              // error(adcY)
+					  fitX->GetParameter(3*nx+2), // meanX
+					  fitY->GetParameter(3*ny+2), // meanY
+					  fitX->GetParError(3*nx+2),  // error(meanX)
+					  fitY->GetParError(3*ny+2),  // error(meanY)
+					  fitX->GetParameter(3*nx+3), // sigmaX
+					  fitY->GetParameter(3*ny+3), // sigmaY
+					  fitX->GetParError(3*nx+3),  // error(sigmaX)
+					  fitY->GetParError(3*ny+3)); // error(sigmaY)
+      if (Debug()) newCluster->Print();
+      hits.getHitVec().push_back(newCluster);
+    }
+  }
+  
+  if (Debug()>3) {
+    canv->Divide(2,2);
+    
+    canv->cd(1);
+    profX[module]->Draw();
+    canv->cd(2);
+    profY[module]->Draw();
+    
+    canv->cd(3);
+    histX->Draw();
+    canv->cd(4);
+    histY->Draw();
+    
+    canv->Modified();
+    canv->Update();
+    canv->Draw();
+    if (nClusX || nClusY) {
+      while (!gSystem->ProcessEvents()){gSystem->Sleep(200);}
+    }
+  }
```

Comment:
So, if someone wants to debug a different maker by running a BFC chain with GMT options, they’ll have to suffer because you decided to draw histograms and introduce artificial delays...